crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -1536,25 +1560,39 @@ impl std::fmt::Display for PubGrubHint {
                     "--index-strategy unsafe-best-match".green(),
                 )
             }
-            Self::UnauthorizedIndex { index } => {
-                write!(
-                    f,
-                    "{}{} An index URL ({}) could not be queried due to a lack of valid authentication credentials ({}).",
-                    "hint".bold().cyan(),
-                    ":".bold(),
-                    index.without_credentials().cyan(),
-                    "401 Unauthorized".red(),
-                )
+            Self::UnauthorizedIndex {
+                index,
+                custom_message,
+            } => {
+                if let Some(message) = custom_message {
+                    write!(f, "{}{} {}", "hint".bold().cyan(), ":".bold(), message,)
+                } else {
+                    write!(
+                        f,
+                        "{}{} An index URL ({}) could not be queried due to a lack of valid authentication credentials ({}).",
+                        "hint".bold().cyan(),
+                        ":".bold(),
+                        index.without_credentials().cyan(),
+                        "401 Unauthorized".red(),
+                    )
+                }
```

Comment:
This is a bit drive-by, as I haven't had time to look at the rest of the pull request in depth, but... it seems wrong to drop the rest of our messaging here. Can you explain the motivation for that? I'd expect the custom messaging to just be an addendum.